File: app/app/src/main/java/com/example/hakonsreader/api/interfaces/SaveableRequest.java
package com.example.hakonsreader.api.interfaces;

/**
 * Interface for requests that offer functionality to save/unsave a post or comment
 */
public interface SaveableRequest {

    /**
     * Save a comment or post
     */
    void save(OnResponse<Void> onResponse, OnFailure onFailure);

    /**
     * Save a comment or post
     */
    void unsave(OnResponse<Void> onResponse, OnFailure onFailure);

}

File: app/app/src/main/java/com/example/hakonsreader/api/interfaces/SaveableRequest.kt
package com.example.hakonsreader.api.interfaces

import com.example.hakonsreader.api.model.RedditListing
import com.example.hakonsreader.api.responses.ApiResponse

/**
 * Interface for a [RedditListing] that can be saved/unsaved
 */
interface SaveableRequest {

    /**
     * Save the listing
     *
     * @return No response data is returned
     */
    suspend fun save() : ApiResponse<Any?>

    /**
     * Unsave the listing
     *
     * @return No response data is returned
     */
    suspend fun unsave() : ApiResponse<Any?>
}